addons/website_event/controllers/main.py

```diff
@@ -139,12 +140,12 @@ def dom_without(without):
             scope=5)
 
         order = 'date_begin'
-        if searches.get('date', 'all') == 'old':
+        if searches.get('date', 'upcoming') == 'old':
             order = 'date_begin desc'
         order = 'is_published desc, ' + order
         events = Event.search(dom_without("none"), limit=step, offset=pager['offset'], order=order)
 
-        keep = QueryURL('/event', **{key: value for key, value in searches.items() if (key == 'search' or value != 'all')})
+        keep = QueryURL('/event', **{key: value for key, value in searches.items() if (key == 'search' or (value != 'upcoming' if key == 'date' else value != 'all'))})
```

Comment:
could be the opportunity to split in multi lines. Because it begins to become :exploding_head:

Reply:
Hello @dbeguin 

Changes Done.

Thanks